src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1422,6 +1421,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return false;
             }
 
+            var expressionBody = GetExpressionBodyFromMember(containingMember);
+            if (expressionBody != null)
+            {
+                return ArrowExpressionClauseExpressionAdjustedSpan(expressionBody).IntersectsWith(position);
```

Comment:
This only handles at the very beginning of the expression bodied member.  Would a better check be to see if the token on the left is the =>?